etl/src/main/java/org/pih/hivmigration/etl/sql/ProgramMigrator.groovy

```diff
@@ -0,0 +1,252 @@
+            SELECT source_patient_id, health_center, enrollment_date, art_start_date, outcome_date, outcome FROM hivmigration_programs_raw
+            WHERE starting_health_center IS NULL
+            ;
+        ''')
+
+        executeMysql("Load initial enrollments into hivmigration_programs table", '''
+            insert into hivmigration_programs
+                (source_patient_id, location_id, enrollment_date, art_start_date, outcome_date, outcome)
+            SELECT
+                source_patient_id,
+                starting_health_center,
+                enrollment_date,
+                IF (art_start_date < health_center_transfer_date, art_start_date, NULL),
+                health_center_transfer_date,
+                'TRANSFERRED_INTERNALLY'
+            FROM hivmigration_programs_raw
+            WHERE starting_health_center IS NOT NULL
+            ;
+        ''')
+
+        executeMysql("Load current enrollments into hivmigration_programs table", '''
+            insert into hivmigration_programs
+                (source_patient_id, location_id, enrollment_date, art_start_date, outcome_date, outcome)
+            SELECT
+                source_patient_id,
+                health_center,
+                health_center_transfer_date,
+                IF (art_start_date < health_center_transfer_date, health_center_transfer_date, art_start_date),
+                outcome_date,
+                outcome
+            FROM hivmigration_programs_raw
+            WHERE starting_health_center IS NOT NULL
+            ;
+        ''')
+
+        executeMysql("Load to patient_program table", '''
+            insert into patient_program
+                (patient_program_id, patient_id, program_id, date_enrolled, date_completed, location_id, outcome_concept_id, creator, date_created, uuid)
+            select
+                h.patient_program_id, 
+                p.person_id,
+                2,
+                h.enrollment_date, 
+                h.outcome_date, 
+                hc.openmrs_id,
+                o.outcome_concept_id, 
+                1,
+                date_format(curdate(), '%Y-%m-%d %T'),
+                uuid()
+            from 
+                hivmigration_programs h 
+            inner join 
+                hivmigration_patients p on h.source_patient_id = p.source_patient_id 
+            left outer join
+                hivmigration_health_center hc on h.location_id = hc.hiv_emr_id
+            left outer join 
+                hivmigration_outcome o on h.outcome = o.outcome
+            ; 
+        ''')
+        
+        executeMysql("Add patient state", '''
+            insert into patient_state
+                (patient_program_id, state, start_date, end_date, creator, date_created, uuid)
+            select
+                patient_program_id,
+                if (art_start_date is null or art_start_date > enrollment_date, 2, 1),
```

Comment:
We should not hard code primary key state ids here (2 and 1), but should look these up by uuid